### PR TITLE
ESQL: Add syntax clarifications on identifiers and literals

### PR DESCRIPTION
**Pull request by bpintea**
Document the syntax for identifiers and literals.

### CODE REVIEW

File: docs/reference/esql/esql-syntax.asciidoc [hunk 1]
```diff
@@ -36,6 +36,65 @@ source-command | processing-command1 | processing-command2
...
+[[esql-identifiers]]
+==== Identifiers
+
+Unless containing special characters, the identifiers can be used as-are and
+don't require quoting. If they do, however, they must be quoted with
+backticks (```). The special characters are those that have already a semantic
+meaning for a given command, like arithmetic operators (`-`, `+` etc.),
+decimals delimiters (`.`), field delimiters (`,`), command separators (`|`),
+subfield referencing (`.`) a.s.o. This is also applicable to identifiers that
+could be interpreted like numbers (the scientific notation, like `1e2`) or
```
Comment (astefan) [inline docs/reference/esql/esql-syntax.asciidoc:49]:
```suggestion
could be interpreted as numbers (for example, the scientific notation `1e2`) or
```

File: docs/reference/esql/esql-syntax.asciidoc [hunk 2]
```diff
@@ -36,6 +36,65 @@ source-command | processing-command1 | processing-command2
...
+// Copy one field
+FROM index
+| EVAL my_field = `1.field`
+----
+
+[discrete]
+[[esql-literals]]
+==== Literals
+
+{esql} supports currently numeric and string literals. The numeric literals
```
Comment (astefan) [inline docs/reference/esql/esql-syntax.asciidoc:75]:
```suggestion
{esql} currently supports numeric and string literals. The numeric literals
```

File: docs/reference/esql/esql-syntax.asciidoc [hunk 3]
```diff
@@ -36,6 +36,65 @@ source-command | processing-command1 | processing-command2
...
+can be used as are. The string literals need to be quoted in double
```
Comment (astefan) [inline docs/reference/esql/esql-syntax.asciidoc:76]: `as they are` or `as is`. I believe `as are` is incorrect.

File: docs/reference/esql/esql-syntax.asciidoc [hunk 4]
```diff
@@ -36,6 +36,65 @@ source-command | processing-command1 | processing-command2
 ----
 ====
 
+[discrete]
+[[esql-identifiers]]
+==== Identifiers
+
+Unless containing special characters, the identifiers can be used as-are and
```
Comment (astefan) [inline docs/reference/esql/esql-syntax.asciidoc:43]:
```suggestion
Unless containing special characters, the identifiers can be used as is and
```

File: docs/reference/esql/esql-syntax.asciidoc [hunk 5]
```diff
@@ -36,6 +36,65 @@ source-command | processing-command1 | processing-command2
...
+don't require quoting. If they do, however, they must be quoted with
+backticks (```). The special characters are those that have already a semantic
+meaning for a given command, like arithmetic operators (`-`, `+` etc.),
+decimals delimiters (`.`), field delimiters (`,`), command separators (`|`),
+subfield referencing (`.`) a.s.o. This is also applicable to identifiers that
```
Comment (astefan) [inline docs/reference/esql/esql-syntax.asciidoc:48]: `a.s.o.`? Maybe provide all of them?

**Review by astefan — APPROVED** [review]
LGTM with few comments.

File: docs/reference/esql/esql-syntax.asciidoc [hunk 6]
```diff
@@ -36,6 +36,65 @@ source-command | processing-command1 | processing-command2
 ----
 ====
 
+[discrete]
+[[esql-identifiers]]
+==== Identifiers
+
+Unless containing special characters, the identifiers can be used as-are and
+don't require quoting. If they do, however, they must be quoted with
+backticks (```). The special characters are those that have already a semantic
```
Comment (abdonpijpelink) [inline docs/reference/esql/esql-syntax.asciidoc:45]:
```suggestion
backticks (+{backtick}+). The special characters are those that have already a semantic
```
Getting a single backtick in an inline code block is not trivial.

File: docs/reference/esql/esql-syntax.asciidoc (hunk 5 above)
Comment (bpintea) [reply docs/reference/esql/esql-syntax.asciidoc]: I've reworked this paragraph to simplify it.

File: docs/reference/esql/esql-syntax.asciidoc (hunk 4 above)
Comment (bpintea) [reply docs/reference/esql/esql-syntax.asciidoc]: Thanks.

File: docs/reference/esql/esql-syntax.asciidoc (hunk 3 above)
Comment (bpintea) [reply docs/reference/esql/esql-syntax.asciidoc]: Reworked this paragraph too, splitting strings and numeric literals.

File: docs/reference/esql/esql-syntax.asciidoc (hunk 6 above)
Comment (bpintea) [reply docs/reference/esql/esql-syntax.asciidoc]: Nice, thanks!

**Review by abdonpijpelink — APPROVED** [review]
LGTM

File: docs/reference/esql/esql-syntax.asciidoc [hunk 7]
```diff
@@ -36,6 +36,101 @@ source-command | processing-command1 | processing-command2
...
+[[esql-identifiers]]
+==== Identifiers
+
+The identifiers can be used as they are and don't require quoting, unless
+containing special characters, in which case they must be quoted with
+backticks (+{backtick}+). What "special characters" means is command dependent.
+
+For <<esql-from, FROM>>, <<esql-keep, KEEP>>, <<esql-drop, DROP>>,
+<<esql-rename, RENAME>>, <<esql-mv_expand, MV_EXPAND>> and
+<<esql-enrich, ENRICH>> these are: `=`, ```, `,`, ` ` (space), `|` ,
```
[the remark this answers is not in the document]
Comment (bpintea) [reply docs/reference/esql/esql-syntax.asciidoc]: good catch, thanks.

**Pull request merged**